ocs_ingester/ingester.py

```diff
@@ -252,6 +252,10 @@ def ingest(self):
         # Upload the file to s3 and get version information back
         version = self.s3.upload_file(self.file, fits_dict)
 
+        # Remove dashes from version_set key to comply with science archive format
+        if 'key' in version:
+            version['key'] = version['key'].replace('-', '')
```

Comment:
I'm not convinced this should go here since it would affect all uploads, both for minio and for s3, and the reason this is here is to get minio to work, and removing dashes happens to work for that case... also it's not something that future me would expect and would probably be confused why it is being done - amazon docs say that [they always generate the IDs for you and they can not be edited](https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html) which makes this unexpected to me. Do you know whether version IDs generated by s3 will always never have dashes? If not, I would lean towards [ingesting the file one step at a time](https://ingester.readthedocs.io/en/latest/documentation.html#module-ingester) in the example project and removing dashes from the version ID there. That or at least making it configurable whether to remove characters from the generated version ID.

Reply:
Oh, doing it in the ingestion script makes so much more sense. I'll do that and then remove this bit of code.